symphony/lib/toolkit/class.emailgateway.php

```diff
@@ -610,13 +631,15 @@ protected function getSectionAttachments()
                      . $this->contentInfoString($file['mime-type'], $file['file'], $file['filename'], $file['charset'])
                      . EmailHelper::base64ContentTransferEncode($file_content);
             } else {
-                if (!$tmp_file === false) {
-                    $filename = $original_filename;
-                } else {
-                    $filename = $file['file'];
+                if ($this->_ignore_attachment_errors === false) {
+                    if (!$tmp_file === false) {
```

Comment:
Ah, I see. I just wanted to make it "precise", because `!$this->_ignore_attachment_errors` would also be true for a value of `0` or an empty string. But of course my setter function only allows boolean values—so should I use a "less strict" check here?

It might then simply be `if ($this->_validate_attachment_errors)` with the other change you propose. I would use `if ($this->_validate_attachment_errors === true)`. :-)

I like to learn, and you know better than me what is "good practice". So which version is better in this context?